### PR TITLE
wip

### PR DESCRIPTION
**Pull request by dancoombs**
[Closes/Fixes] #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: bin/tools/src/bin/load_chain_spec.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            let file = File::open(file).expect("should open file");
+            serde_yaml::from_reader(file).expect("should parse yaml")
+        }
+        None => OptionalChainSpec::default(),
+    };
+
+    let env_spec = match envy::prefixed("CHAIN_").from_env::<OptionalChainSpec>() {
+        Ok(config) => config,
+        Err(error) => panic!("{:#?}", error),
+    };
```
Comment (dphilipson) [inline bin/tools/src/bin/load_chain_spec.rs:51]: Is this match different/better than calling `.unwrap()`?

File: crates/types/src/chain_spec.rs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        $vis struct $struct_name {
+            $( #[$doc] $fv $field: $ty ),*
+        }
+
+        paste! {
+            $(#[$attr])*
+            #[derive(Deserialize, Default)]
+            $vis struct [<Optional $struct_name>] {
+                /// The base chain spec to use
+                pub base: Option<String>,
```
Comment (dphilipson) [inline crates/types/src/chain_spec.rs:35]: Is this supposed to be here? Is it different from the other fields?

**Review by dphilipson — COMMENTED** [review]
I love it.

File: bin/tools/src/bin/load_chain_spec.rs (hunk 1 above)
Comment (dphilipson) [reply bin/tools/src/bin/load_chain_spec.rs:51]: Also don't forget that you can write stuff like this and the printlns as
```rust
panic!("{error:#?}")
```

Comment (dancoombs) [reply bin/tools/src/bin/load_chain_spec.rs:51]: Lol no it is not, wasn't very clean writing this bin

File: crates/types/src/chain_spec.rs (hunk 2 above)
Comment (dancoombs) [reply crates/types/src/chain_spec.rs:35]: Yeah the idea was that during the optional phase you can define a `base` that merges all configuration from that base configuration into yours, useful for testnets to inherit the config of the mainnet. This field goes away in the actual chain spec config.

**Pull request closed without merging**